main.go

```diff
@@ -59,38 +69,114 @@ func indexHandler(w http.ResponseWriter, r *http.Request) {
 		// Unmarshal JSON
 		b, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			http.Error(w, "Invalid request method", http.StatusBadRequest)
+			raven.CaptureError(err, nil)
+			http.Error(w, "Invalid request", http.StatusBadRequest)
 			return
 		}
 		defer r.Body.Close()
 		req := indexRequest{}
 		json.Unmarshal(b, &req)
 
-		// Convert []Trytes to []Transaction
-		txs := make([]giota.Transaction, len(req.Trytes))
-		for i, t := range req.Trytes {
-			tx, _ := giota.NewTransaction(t)
-			txs[i] = *tx
-		}
+		go attachAndBroadcastToTangle(&req)
+
+		w.WriteHeader(http.StatusNoContent)
+	} else {
+		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
+	}
+}
+
+func attachAndBroadcastToTangle(indexReq *indexRequest) {
+	provider := os.Getenv("PROVIDER")
+	// minDepth, _ := strconv.ParseInt(os.Getenv("MIN_DEPTH"), 10, 64)
+	minWeightMag, _ := strconv.ParseInt(os.Getenv("MIN_WEIGHT_MAGNITUDE"), 10, 64)
 
-		// Get configuration.
-		provider := os.Getenv("PROVIDER")
-		minDepth, _ := strconv.ParseInt(os.Getenv("MIN_DEPTH"), 10, 64)
-		minWeightMag, _ := strconv.ParseInt(os.Getenv("MIN_WEIGHT_MAGNITUDE"), 10, 64)
+	// Convert []Trytes to []Transaction
+	txs := make([]giota.Transaction, len(indexReq.Trytes))
+	for i, t := range indexReq.Trytes {
+		tx, _ := giota.NewTransaction(t)
+		txs[i] = *tx
+	}
+
+	api := giota.NewAPI(provider, nil)
+	// _, pow := giota.GetBestPoW()
```

Comment:
We also don't pass in the `pow` algorithm, which was the whole reason we switched to golang in the first place. I'm not sure if this is decided for us under the hood. I can look and verify.